src/test/java/task/DeadlineTest.java

```diff
@@ -0,0 +1,27 @@
+package task;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+public class DeadlineTest {
+    @Test
+    public void testDescription() {
+        assertEquals("[D][ ] hi (by: Sunday)",new Deadline("hi", "Sunday").getString());
+        assertEquals("[D][ ] bye (by: Dec 12 2020)",new Deadline("bye", "2020-12-12").getString());
+    }
+
+    @Test
+    public void testStorageFormat() {
+        assertEquals("0 deadline hi /by Sunday\n",new Deadline("hi", "Sunday").getStorageFormat());
+        assertEquals("0 deadline bye /by 2020-12-12\n",new Deadline("bye", "2020-12-12").getStorageFormat());
+    }
+
+    @Test
+    public void emptyTodo() throws Exception {
```

Comment:
Perhaps this could be named testEmptyDeadline. Methods are normally named as verbs, and this method tests for an empty Deadline rather than an empty Todo.